_source/_posts/2021-08-09-aws-toolkit-for-visual-studio.md

```diff
@@ -0,0 +1,415 @@
+{% img blog/aws-toolkit-for-visual-studio/get-started-aws-in-vs.png alt:"Visual Studios AWS login prompt" width:"800" %}{: .center-image }
+
+The screen will instruct you to log into the IAM Users page in the AWS console and create a user. Click on that link and set up a new user.
+
+{% img blog/aws-toolkit-for-visual-studio/aws-iam.png alt:"AWS IAM Users page" width:"800" %}{: .center-image }
+
+Create the new user, then attach an administrator or power user policy to the user. Next, you will be presented with the *Access Key* and *Secret Key*. Copy and paste those into the AWS Toolkit, then press save, and close.
+
+## Publish to AWS Elastic Beanstalk
+
+Next, right-click on your project and click **Publish to AWS Elastic Beanstalk**. This will open a new wizard to guide you through the publish process.
+
+First is the application tab. Select *Create a new application environment* and then select *Next*.
+
+{% img blog/aws-toolkit-for-visual-studio/aws-publish-to-beanstalk.png alt:"Publish to AWS Elastic Beanstalk" width:"800" %}{: .center-image }
+
+Under the *Application Environment* screen select the production version of your environment. This action will automatically populate the URL for your Elastic Beanstalk instance. You'll want to check to make sure the URL is available before moving forward.
+
+{% img blog/aws-toolkit-for-visual-studio/aws-application-env.png alt:"Publish to AWS Elastic Beanstalk" width:"800" %}{: .center-image }
+
+Under the *Options* tab, the only change you need to make is to use a t2.micro instance to maintain your free tier status. If you need to scale up later Amazon makes that easy through the AWS Toolkit.
+
+{% img blog/aws-toolkit-for-visual-studio/aws-ec2-config.png alt:"AWS container configuration" width:"800" %}{: .center-image }
+
+Under the options tab you will want to select the *Release* configuration.
+
+{% img blog/aws-toolkit-for-visual-studio/aws-application-options.png alt:"AWS Application Options" width:"800" %}{: .center-image }
+
+Finally, on the view tab, click **Deploy** and wait a few minutes for the site to be published and the server to be live. You will receive a notification when it is ready. From there, you can click the link to your site to see the published version. However, at this time, you will notice your login screen doesn't work as expected. This is because you still need to populate your production Okta credentials.
+
+## Create Your Okta Production Application
+
+Open your CLI again and go through the `okta apps create` wizard. However this time, instead of using the `localhost` domain provided by Visual Studio, replace the domain in your *Redirect URI* and *Post Logout RedirectURI* with the domain that AWS published your application to. After a moment the CLI will again write your details to `.okta.env` and you can replace your `appsettings.Production.json` values with these new ones. Again, only use the domain of your *Issuer* in the domain field of your appsettings file.
```

Comment:
```suggestion
Open your CLI again and go through the `okta apps create` wizard. However this time, instead of using the `localhost` domain provided by Visual Studio, replace the domain in your *Redirect URI* and *Post Logout RedirectURI* with the domain where AWS published your application. After a moment, the CLI will again write your details to `.okta.env` and you can replace your `appsettings.Production.json` values with these new ones. Again, only use the domain of your *Issuer* in the domain field of your appsettings file.
```

Should these be bold too, or italics? – *Redirect URI* and *Post Logout RedirectURI*